### ISSUE TITLE
Missing note

### ISSUE THREAD

**Issue by alltheseas** [nostrability/nostrability#20]
## client initiating action

-

## client receiving or observing action (or lack thereof)

Damus
Plebstr

## what happens 

Sometimes I can see a reply made by someone I follow. However, I cannot see the note being replied to. 

## suggestion

**Comment by staab** [comment]
This is a result of missing relay hints and/or poor relay selection based on the gossip model

**Comment by alltheseas** [comment]
@matthewsvancer does Plebstr implement relay hints or gossip?

**Comment by alltheseas** [comment]
Damus does not implement nip-19 currently.

https://github.com/damus-io/damus/issues/1147

**Comment by jb55** [comment]
from what I can tell this only happens when I'm viewing posts from users who are using primal. I'm guessing they are not broadcasting the note they are replying to, which was damus' hack that fixed most of the problems without needing the gossip model.

**Comment by jb55** [comment]
I still like how simple the broadcast approach is, it is akin to quote-reply in email. You shouldn't assume the client has the means or knowhow to pull the note in a way that other clients might pull it. The lowest common denominator is to broadcast things you are replying to.

This might miss the entire thread, but it's a simple thing that most clients shouldn't have a problem doing.

**Comment by staab** [comment]
Yeah, broadcast is maybe even better than relay hints at helping with this

**Comment by alltheseas** [comment]
@mbrat1 @moysa is Primal broadcasting on reply?

**Comment by alltheseas** [comment]
Now in meme form plus commentary from Dilger and Vitor

https://njump.me/nevent1qqsyuudrdvswfyenzva8wkkekva59j6ew37dl0e99dtpu7cz64r62zspz4mhxue69uhk2er9dchxummnw3ezumrpdejqzxrhwvaz7tmddahxzepwdf3r2dfwvdhk6w3cxqurqqgdwaehxw309ahx7uewd3hkcqgswaehxw309ahx7um5wgh8w6twv5vs38eq

**Comment by alltheseas** [comment]
Official meme of this issue

https://njump.me/nevent1qqsfn588y9arvh9f0vlqlphprzl4dqjgljn3kthwdrg4y2drw7r4e0gpz3mhxue69uhkymrpwd68ytnxxaazu7re0gq3camnwvaz7tmxv4jkguewdehhxarj9e3xzmny9akk2mt9wvq3camnwvaz7tmwdaehgu3wd46hg6tw09mkzmrvv46zucm0d5q3gamnwvaz7tmwdaehgu3wdau8gu3wv3jhvkankxt

**Comment by moysa** [comment]
No, Primal doesn't broadcast the original note on reply. @jb55 are you broadcasting the root note as well as the note that is being replied to?

**Comment by mbrat1** [comment]
All Primal clients have been updated to broadcast the parent note when publishing a reply. Hopefully this does the trick!

**Comment by jb55** [comment]
On Thu, Feb 22, 2024 at 08:00:24AM -0800, mbrat1 wrote:
>All Primal clients have been updated to broadcast the parent note when
>publishing a reply. Hopefully this does the trick!

awesome! thanks guys.

**Comment by jb55** [comment]
> No, Primal doesn't broadcast the original note on reply. @jb55 are you broadcasting the root note as well as the note that is being replied to?

not the root, just the direct reply. doing the root as well would be even better.

**Comment by benthecarman** [comment]
@mbrat1 @moysa replying to notes feels slower to me, you guys might be waiting on the rebroadcast

**Comment by mbrat1** [comment]
Yeah that was totally it. We are no longer doing that; replies should be faster now.

**Comment by alltheseas** [comment]
Still seeing missing note. I think poster is using Primal. Is it possible it's an older version of Primal? 

![image](https://github.com/nostrability/nostrability/assets/64376233/a5ce5197-4373-4692-bf33-c6e69546812a)

https://njump.me/nevent1qqsg7fc6sydrz2mmzca6ccdd5g324xwnecedq5j5kn3t7uyg9df6l8gvnksu9

**Comment by mbrat1** [comment]
Yep, Preston is on Primal and he is using the latest version (spoke to him earlier today).

**Comment by alltheseas** [comment]
@jb55 if the broadcast root note did not fix the missing note issue, what can we try next?
